lib/middleware/routes-output/routes-output.js

```diff
@@ -96,32 +82,45 @@ const generateOutput = async (req, res) => {
     let output = await nunjucksConfiguration.renderPage(pageInstance, Object.assign({}, res.locals)) // getPageOutput()
     output = output.replace(/<head>/, `<head><base href="http://localhost:${PORT}">`)
     output = output.replace(/<title>.*<\/title>/, `<title>${submission}</title>`)
-    // const context = await browser.createIncognitoBrowserContext()
-    let page
+
+    const browser = await puppeteer.launch({
+      args: [
+        '--no-sandbox',
+        '--disable-setuid-sandbox',
+        '--disable-dev-shm-usage'
+      ]
+    })
+
+    let pdfOutput
+    let browserProcessID
+
     try {
-      page = await browser.newPage()
+      browserProcessID = browser.process().pid
```

Comment:
surely if this block errors, we never get the pid and so the fkill will silently fail?